### DISCUSSION TITLE
Wierd behavior with web search prompt generation

### DISCUSSION THREAD

**Discussion by mightykatun** [open-webui/open-webui#7232] [Issues]
I've recently observed a strange behavior with web search prompt generation that seems to occur across the board with all models. In short, when starting a chat by asking for a specific thing: news for example, the model generates a relevant search prompt, however, when asked another question, like the weather, it generates a news-related prompt again (see screenshots).

Could it be that the prompt for search queries remains always the same?

![image](https://github.com/user-attachments/assets/61f29f27-3065-4d74-9e6d-13422c5310fe)\
First news related query.

![image](https://github.com/user-attachments/assets/72af94c5-eeab-4aa3-a263-79d10625dfe6)\
Then weather.

Those two search queries seem to be generated by the same prompt (the news related one)

**Comment by mightykatun** [comment]
The issue seems to be model-dependant or something similar, as it does not happen all the time, the query has trouble if you change subject abruptly, but not always strangely enough.

**Reply by mikygit** [reply]
Same problem here.
What you say makes sense although it means that the RAG web search is not really usable as it is :-(

**Comment by mightykatun** [comment]
It seems like this is not a problem anymore, at least not always, sometimes the models stay stuck with the same prompt. I think the generation has improved though
